Respond to ping6 packets in on-a-stick mode

dpino opened the pull request:
Fixes #709.

The lwAFTR was ignoring VLAN tag settings when running in on-a-stick mode. That can be inferred from #709 stacktrace:

```
apps/lwaftr/ipv6_apps.lua:145: Could not resolve IPv6 address: 2003:1b0b:fffd:831::1
stack traceback:
	core/main.lua:149: in function <core/main.lua:147>
```

To confirm that, I run a ping6 test with VLAN tags to the lwAFTR in on-a-stick mode. Indeed, there were no replies. 

The problem was that a function necessary for VLAN set up in on-a-stick mode was using the former configuration naming. Fixed.

teknico commented on src/program/lwaftr/run/run.lua:
```diff
@@ -124,9 +124,10 @@ end
-   if not opts["on-a-stick"] then return false end
-   if not conf.vlan_tagging then return true end
-   return conf.v4_vlan_tag == conf.v6_vlan_tag
+   if not opts["on-a-stick"] then return true end
```
This inverts the previous behavior, and is not in accordance with the above comment anymore. Who's right, the comment or the code?

dpino replied:
Good catch, that was a non intended change.

teknico commented on src/program/lwaftr/run/run.lua:
```diff
@@ -124,9 +124,10 @@ end
-   return conf.v4_vlan_tag == conf.v6_vlan_tag
+   if not opts["on-a-stick"] then return true end
+   local internal_interface = conf.softwire_config.internal_interface
+   local external_interface = conf.softwire_config.external_interface
```
We only use these two locals once, are they useful? Can't we just say this?
```
return conf.softwire_config.internal_interface.vlan_tag == conf.softwire_config.external_interface.vlan_tag
```

teknico replied:
Or maybe:
```
local softwire_config = conf.softwire_config
return softwire_config.internal_interface.vlan_tag == softwire_config.external_interface.vlan_tag
```

dpino replied:
Yep, although as the path is so long it breaks the 80-column wrapping, in theory the codebase should respect. Other snippets of code cache these variables too, such as https://github.com/Igalia/snabb/blob/lwaftr/src/program/lwaftr/setup.lua#L33 I followed the same pattern.

teknico replied:
In that other case there are many uses of these locals, so they are useful.

In this case, my second suggestion at least has two uses of `softwire_config`. :-)

True, the resulting line is longer than 80 chars. Do we have guidelines? The codebase has many lines longer than that.

dpino replied:
Yes, Snabb codebase has guidelines. There used to be a document with these guidelines, which mostly follow the Lua guidelines, but I cannot find it anymore. One of the guidelines was the 80-colum wrap, which most of the codebase follows, although there's code that don't.

teknico reviewed (commented):
Not clear about the new behavior.

dpino committed:
Respond to ping6 packets in on-a-stick mode

Adapt function requires_splitter to new configuration naming.

dpino committed:
Add test ping lwAFTR's V4 and V6 interfaces from a Linux kernel interface

lwAFTR runs in on-a-stick mode.

dpino commented:
Added a test to test pings from a Linux kernel interface. Example:

```bash
$ sudo SNABB_PCI0=0000:03:00.1 SNABB_PCI1=0000:83:00.1 program/lwaftr/tests/hw/selftest.sh 
Success: Ping to internal interface
Success: Ping to external interface
```

teknico commented on src/program/lwaftr/run/run.lua:
```diff
@@ -121,12 +121,15 @@ function parse_args(args)
+      local external_interface = conf.softwire_config.external_interface
+      return internal_interface.vlan_tag == external_interface.vlan_tag
+   end
+   return false
```
Thank you, much clearer this way.

teknico reviewed (approved):
Thanks for adding the test too. LGTM.

dpino committed:
Add configuration file

dpino committed:
Skip hw ping to lwAFTR test in CI

The program/lwaftr/tests/hw/test_ping_on_a_stick.sh test cannot be run
inside a container for several reasons:

- The test contains commands, such as iproute2 or lshw,  not supported
in the default image.
- The test binds a NIC to the Linux kernel by PCI address. This enables
the NIC interface. However, when run inside the container, the card gets
rebound to the kernel, but the interface doesn't appear inside the
container but the host. Without an addressable kernel-managed interface,
it's not possible to run the test.

dpino commented:
The test cannot be run inside a container for several reasons, the most important is not being able to enable a kernel-managed interface inside the container. Fixing this issue will require more investigation work.

The test works when launched manually and it's worth to keep it to test a lwAFTR responds to pings from other devices successfully.
